src/heroic.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::HashMap;
+
+use crate::{
+    config::RootsConfig,
+    manifest::{Manifest, Store},
+    prelude::{normalize_title, StrictPath},
+};
+
+// Deserialization of Heroic gog_store/installed.json
+#[derive(serde::Deserialize)]
+struct HeroicInstalledGame {
+    #[serde(rename = "appName")]
+    game_id: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct HeroicInstalled {
+    installed: Vec<HeroicInstalledGame>,
+}
+
+// Deserialization of Heroic gog_store/library.json
+#[derive(serde::Deserialize)]
+struct GogLibraryGame {
+    app_name: String,
+    title: String,
+}
+#[derive(serde::Deserialize)]
+struct GogLibrary {
+    games: Vec<GogLibraryGame>,
+}
+
+// Deserialization of Legendary legendary/installed.json
+#[derive(serde::Deserialize)]
+struct LegendaryInstalledGame {
+    #[serde(rename = "app_name")]
+    game_id: String,
+    title: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct LegendaryInstalled(HashMap<String, LegendaryInstalledGame>);
+
+// TODO.2022-10-10 heroic: windows location for legendary
+// TODO.2022-10-15 heroic: make relative to heroic root in question!
+const LEGENDARY_PATHS: &[&str] = &[
+    "~/.config/legendary",
+    // TODO.2022-10-20 heroic: flatpak install is not supported yet
+    // "~/.var/app/com.heroicgameslauncher.hgl/config/legendary",
+];
+
+#[derive(Clone, Default, Debug)]
+pub struct HeroicGames {
+    games: HashMap<String, StrictPath>,
```

Comment:
I agree.  Speaking from my experience with having two installations of heroic by accident (didn't properly remove one of them) this causes confusion.

Reply:
I wonder: if someone does that (switch from standalone to flatpak or vice versa), how should ludusavi react on restore?

Reply:
The user could set up a redirect to migrate during restoration if they'd like. Ludusavi would still restore to the old location by default, though, since we don't want to move files around unless the user asks for it.

Reply:
Though an edge case, it might be worth documenting it somewhere.  Do you plan to add a wiki to the repository or would the README.md a better place for that?

Reply:
I can add a note to the README. There's already a section on redirects, so I could add a portion to cover this specific kind of use case.

Reply:
sounds good!